io/plugins/eu.esdihumboldt.hale.io.jdbc/src/eu/esdihumboldt/hale/io/jdbc/JDBCTableCollection.java

```diff
@@ -304,8 +304,20 @@ public int size() {
 
 			return count;
 		} catch (SQLException e) {
-			log.warn("Could not determine query size", e);
-			return UNKNOWN_SIZE;
+			log.warn("Could not determine query size by count query");
+			try (Connection connection = createConnection()) {
+				Statement st = connection.createStatement();
+				ResultSet res = st.executeQuery(sqlQuery);
+				if(res.next()) {
```

Comment:
Please review the formatting (GitHub doesn't do that automatically if you edit via the browser)